H2TauTau/scripts/multiHarvest.py

```diff
@@ -53,63 +50,46 @@ def multithreadmap(f,X,ncores=20, **kwargs):
+    return matched_dirs
+
 options, args = get_options()
-os.system('xrdfs lyogrid06.in2p3.fr ls /dpm/in2p3.fr/home/cms/data/store/user/{}/heppyTrees/{}/{}/ | grep {} > files.out'.format(options.username, options.cmssw_version, options.prod_label, options.cut_on_sample_names))
 
-# Select directories
-selected_fcompos = set()
-print ''
-print 'Selecting directories: '+os.popen('cat files.out | wc -l').readline()[:-1]+' directories to process.'
-Ndir = 0
-with open('files.out') as f:
-    for line in f.readlines():
-        Ndir += 1
-        print '  '+str(Ndir)+'/'+os.popen('cat files.out | wc -l').readline()[:-1]
-        component = line[line.rfind('/')+1:-1]
-        os.system('xrdfs lyogrid06.in2p3.fr ls /dpm/in2p3.fr/home/cms/data/store/user/gtouquet/heppyTrees/CMSSW_9_4_11_cand1/diTau_2018_modular_cfg/{}/ > {}.out'.format(component,component))
-        with open('{}.out'.format(component)) as fcomponent:
-            cutstring = None
-            if options.select_date == 'R' :
-                matching_fcompo = [fcomponent.readlines()[-1]]
-            else :
-                matching_fcompo = [line for line in fcomponent.readlines() if options.select_date in line]
-                while len(matching_fcompo) > 1:
-                    print '    Warning, several matching directories for component '+component+':'
-                    for compo in matching_fcompo :
-                        print '     '+compo[:-1]
-                    cutstring = raw_input('    Please enter a cut string on these names to select only one or pass all to get them all:')
-                    print ''
-                    if cutstring == 'all' :
-                        for fcompo in matching_fcompo :
-                            selected_fcompos.add(fcompo)
-                        matching_fcompo = []
-                    else :
-                        matching_fcompo = [line for line in matching_fcompo if cutstring in line]
-            if cutstring is not 'all' :
-                if len(matching_fcompo) == 0 :
-                    print '  Warning, no matching files for component '+component
-                else :
-                    selected_fcompos.add(matching_fcompo[0])
-            os.system('rm '+component+'.out')
-os.system('rm files.out')
+matching_prod_label_dirs = find_dirs_in_dir('/dpm/in2p3.fr/home/cms/data/store/user/{}/heppyTrees/{}/'.format(options.username, options.cmssw_version), to_match=options.prod_label)
+
+selected_dirs = []
+for prod_label_dir in matching_prod_label_dirs:
+    matching_comp_dirs = find_dirs_in_dir(prod_label_dir, to_match=options.cut_on_sample_names)
+    for comp_dir in matching_comp_dirs:
+        matched_dirs = find_dirs_in_dir(comp_dir)
+        if options.select_date == 'R' :
+            matched_dirs = [matched_dirs[-1]]
+        else :
+            matched_dirs = [dirname for dirname in matching_comp_dirs if options.select_date in line]
+        selected_dirs.extend(matched_dirs)
```

Comment:
Any risk to get a directory twice? If so, use a set instead of a list.

Reply:
I don't see how we could get the same directory twice, could you give me an example?

Reply:
Ok when looping over a list given by a ls command.